core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java

```diff
@@ -390,12 +393,59 @@ public void setFeedIdentifier(String feedIdentifier) {
         this.feedIdentifier = feedIdentifier;
     }
 
-    public String getPaymentLink() {
-        return paymentLink;
+    public ArrayList<FeedFunding> extractPaymentLinks(String payLinks) {
+        if (StringUtils.isBlank(payLinks)) {
+            return null;
+        }
+        // old format before we started storing the urls as pay= and fund=
+        ArrayList<FeedFunding> funding = new ArrayList<FeedFunding>();
+        if (!payLinks.contains(FeedFunding.SUPPORT_INTERNAL_SPLIT)
+                && !payLinks.contains(FeedFunding.SUPPORT_INTERNAL_EQUAL)) {
+            funding.add(new FeedFunding(payLinks, ""));
+            return funding;
+        }
+        String [] list = payLinks.split(FeedFunding.SUPPORT_INTERNAL_SPLIT);
+        if (list.length == 0) {
+            return null;
+        }
+
+        for (String str : list) {
+            String [] linkContent = str.split(FeedFunding.SUPPORT_INTERNAL_EQUAL);
+            if (StringUtils.isBlank(linkContent[0])) {
+                continue;
+            }
+            String url = linkContent[0];
+            String title = "";
+            if (linkContent.length > 1 && ! StringUtils.isBlank(linkContent[1])) {
```

Comment:
Please remove the space

```suggestion
            if (linkContent.length > 1 && !StringUtils.isBlank(linkContent[1])) {
```